components/kyma-environment-broker/internal/process/upgrade_kyma/initialisation.go

```diff
@@ -171,6 +183,33 @@ func (s *InitialisationStep) configureKymaVersion(operation *internal.UpgradeKym
 	return nil
 }
 
+func (s *InitialisationStep) performRuntimeChecks(operation internal.UpgradeKymaOperation, instance *internal.Instance, log logrus.FieldLogger) (internal.UpgradeKymaOperation, time.Duration, error) {
+	var errStatus error
+	operation, delay, err := s.checkRuntimeStatus(operation, instance, log)
+	inMaintenance := s.internalEvalUpdater.assistant.GetEvalStatus(operation.Avs) == avs.StatusMaintenance
+
+	// ensures that required pre- and post- logic is executed
+	if operation.State == orchestrationExt.InProgress {
+		// set maintenance evaluation state on init
+		if !inMaintenance {
+			operation, _, errStatus = s.internalEvalUpdater.SetStatusToEval(avs.StatusMaintenance, operation, log)
+			if errStatus != nil {
+				err = errors.Wrap(err, errStatus.Error())
+			}
+		}
+	} else if operation.State == orchestrationExt.Succeeded || operation.State == orchestrationExt.Failed {
+		// restore previous evaluation state on finish
+		if inMaintenance {
+			operation, _, errStatus = s.internalEvalUpdater.RestoreStatusToEval(operation, log)
```

Comment:
So since the AVS statuses are introduced with this PR, the `Original` and `Current` statuses doesn't exist in DB for existing operations records. As a result, JSON deserialization would create the `UpgradeKymaOperation` objects with empty (`""`) AVS status values, and you would restore the empty status both in the DB record and when making the AVS API call which would most probably result in `500 Internal Server Error`. 

I'm thinking of adding a small logic to obtain the `Current` status from AVS through API call when it's empty in DB.